kernel/page_table_entry/src/lib.rs

```diff
@@ -82,9 +93,19 @@ impl PageTableEntry {
 
     /// Extracts the value of the frame referred to by this page table entry.
     fn frame_value(&self) -> Frame {
-        let mut frame_paddr = self.0 as usize;
-        frame_paddr.set_bits(0 .. (PAGE_SHIFT as u8), 0);
-        Frame::containing_address(PhysicalAddress::new_canonical(frame_paddr))
+        #[cfg(target_arch = "x86_64")]
+        {
+            let mut frame_paddr = self.0 as usize;
+            frame_paddr.set_bits(0 .. (PAGE_SHIFT as u8), 0);
+            Frame::containing_address(PhysicalAddress::new_canonical(frame_paddr))
```

Comment:
I changed this in a recent PR to be the same as the aarch64 code, so this function should now be completely architecture independent. Go check out the latest code to see what i mean.

Reply:
see previous reply